packages/core/src/components/forms/numericInput.tsx

```diff
@@ -227,20 +236,19 @@ export class NumericInput extends AbstractPureComponent2<HTMLInputProps & INumer
 
         // in controlled mode, use props.value
         // in uncontrolled mode, if state.value has not been assigned yet (upon initial mount), use props.defaultValue
-        const value = props.value?.toString() ?? state.value;
+        const value = NumericInput.getValue(props, state.value);
```

Comment:
why this change? this function is only used once... I'd prefer to leave it as it was

Reply:
I extracted it into a function, but then I noticed that it was not helping, I forgot to revert